datafusion/sql/src/planner.rs

```diff
@@ -249,14 +267,29 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         Self::new_with_options(context_provider, ParserOptions::default())
     }
 
+    /// add an user defined planner
+    pub fn with_user_defined_planner(
+        mut self,
+        planner: Arc<dyn UserDefinedPlanner>,
+    ) -> Self {
+        self.planners.push(planner);
+        self
+    }
+
     /// Create a new query planner
     pub fn new_with_options(context_provider: &'a S, options: ParserOptions) -> Self {
         let normalize = options.enable_ident_normalization;
+        let array_planner =
+            Arc::new(ArrayFunctionPlanner::try_new(context_provider).unwrap()) as _;
+
         SqlToRel {
             context_provider,
             options,
             normalizer: IdentNormalizer::new(normalize),
+            planners: vec![],
```

Comment:
why not just put the `array_planner` in the vec?

Reply:
My thinking was that I eventually wanted to move the definition into a different crate `datafusion-functions-array`